File: src/services/autostart.ts
import { invoke } from '@tauri-apps/api/core'

export async function setAutoStart(enable: boolean): Promise<boolean> {
  return await invoke<boolean>('set_auto_start', { enable })
}

export async function getAutoStart(): Promise<boolean> {
  return await invoke<boolean>('get_auto_start')
}

File: src-tauri/src/commands/autostart.rs
use tauri::command;

#[command]
pub async fn set_auto_start(
    manager: tauri::State<'_, tauri_plugin_autostart::AutoLaunchManager>,
    enable: bool,
) -> Result<bool, String> {
    if enable {
        manager.enable().map_err(|e| e.to_string())?;
    } else {
        manager.disable().map_err(|e| e.to_string())?;
    }
    Ok(enable)
}

#[command]
pub async fn get_auto_start(
    manager: tauri::State<'_, tauri_plugin_autostart::AutoLaunchManager>,
) -> Result<bool, String> {
    manager.is_enabled().map_err(|e| e.to_string())
}
